src/Microsoft.NetCore.Analyzers/Core/Security/UseSharedAccessProtocolHttpsOnly.cs

```diff
@@ -0,0 +1,140 @@
+                {
+                    var owningSymbol = operationBlockStartContext.OwningSymbol;
+                    var interproceduralAnalysisConfig = InterproceduralAnalysisConfiguration.Create(
+                                                            operationBlockStartContext.Options,
+                                                            SupportedDiagnostics,
+                                                            defaultInterproceduralAnalysisKind: InterproceduralAnalysisKind.None,
+                                                            cancellationToken: operationBlockStartContext.CancellationToken,
+                                                            defaultMaxInterproceduralMethodCallChain: 1);
+
+                    operationBlockStartContext.RegisterOperationAction(operationAnalysisContext =>
+                    {
+                        var invocationOperation = (IInvocationOperation)operationAnalysisContext.Operation;
+                        var methodSymbol = invocationOperation.TargetMethod;
+
+                        if (methodSymbol.Name != "GetSharedAccessSignature")
+                        {
+                            return;
+                        }
+
+                        var namespaceSymbol = methodSymbol.ContainingNamespace;
+
+                        if (namespaceSymbol == null)
+                        {
+                            return;
+                        }
+
+                        var namespaceQualifiedName = namespaceSymbol.ToDisplayString();
+
+                        if (namespaceQualifiedName != "Microsoft.WindowsAzure.Storage" &&
+                            !namespaceQualifiedName.StartsWith("Microsoft.WindowsAzure.Storage.", StringComparison.Ordinal))
+                        {
+                            return;
+                        }
+
+                        var valueContentAnalysisResultOpt = ValueContentAnalysis.GetOrComputeResult(
+                                                                                    invocationOperation.GetTopmostParentBlock().GetEnclosingControlFlowGraph(),
+                                                                                    owningSymbol,
+                                                                                    wellKnownTypeProvider,
+                                                                                    interproceduralAnalysisConfig,
+                                                                                    out var copyAnalysisResult,
+                                                                                    out var pointsToAnalysisResult);
+                        var typeSymbol = methodSymbol.ContainingType;
+
+                        if (typeSymbol.Name != "CloudStorageAccount")
+                        {
+                            var protocolsArgumentOperation = invocationOperation.Arguments.FirstOrDefault(s => s.Parameter.Name == "protocols");
```

Comment:
>s.Parameter.Name == "protocols [](start = 111, length = 30)

Also check the type, as much as you reasonably can.

Reply:
Possibly also check the method symbol matches the expected method signature (return type, parameters length, type and/or name of each expected parameter, etc.)

Reply:
But Semmle just finds the Class in Microsoft.WindowsAzure.Storage namespace but CloudStorageAccount, then check out if a method named GetSharedAccessSignature has a parameter protocols. It doesn't have other condition.

Reply:
That query may only check only an argument named protocols, but I think it'd worth checking that the parameter's Type.TypeKind == TypeKind.Enum, since that's all your code handles.

Up to you whether to find all the specific classes and method symbols.  I think this approach is fine as it means we don't have to update the analyzer the Azure library adds more stuff.